selenium/che-selenium-core/bin/webdriver.sh

```diff
@@ -820,6 +821,65 @@ prepareToFirstRun() {
     checkIfProductIsRun
     cleanUpEnvironment
     initRunMode
+
+    if [[ ${CHE_MULTIUSER} == false ]]; then
+      prepareTestUserForSingleuserChe
+    else
+      prepareTestUserForMultiuserChe
+    fi
+}
+
+prepareTestUserForSingleuserChe() {
+    export CHE_ADMIN_EMAIL=
+    export CHE_ADMIN_PASSWORD=
+
+    export CHE_TESTUSER_EMAIL=che@eclipse.org
+    export CHE_TESTUSER_PASSWORD=secret
+}
+
+prepareTestUserForMultiuserChe() {
+    export CHE_ADMIN_NAME=${CHE_ADMIN_NAME:-admin}
+    export CHE_ADMIN_EMAIL=${CHE_ADMIN_EMAIL:-admin@admin.com}
+    export CHE_ADMIN_PASSWORD=${CHE_ADMIN_PASSWORD:-admin}
+
+    if [[ -n ${CHE_TESTUSER_EMAIL+x} ]] && [[ -n ${CHE_TESTUSER_PASSWORD+x} ]]; then
+        return
+    fi
+
+    # create test user by executing kcadm.sh commands inside keycloak docker container if there are no its credentials among environment variables
+    if [[ "${PRODUCT_HOST}" == "$(detectDockerInterfaceIp)" ]] || [[ "${CHE_SELENIUM_INFRASTRUCTURE}" == "openshift" ]]; then
+        local time=$(date +%s)
+        export CHE_TESTUSER_NAME=${CHE_TESTUSER_NAME:-user${time}}
+        export CHE_TESTUSER_EMAIL=${CHE_TESTUSER_EMAIL:-${CHE_TESTUSER_NAME}@1.com}
+        export CHE_TESTUSER_PASSWORD=${CHE_TESTUSER_PASSWORD:-${time}}
+
+        if [[ "${CHE_SELENIUM_INFRASTRUCTURE}" == "openshift" ]]; then
```

Comment:
`CHE_SELENIUM_INFRASTRUCTURE` was used only to use specific recipe for tests against che on openshift but now we don't need it anymore, but we need some flag which will allow us to do other some infra related things like specific test for che on OS. So can we rename `CHE_SELENIUM_INFRASTRUCTURE` => `CHE_INFRASTRUCTURE` and make sure that this property will allow us to do some infra specific actions.